### PR TITLE
Adding Arabic Dates Classification using DL #329

### LINKED ISSUE #329: 📃: Arabic Dates Classification using DL
:red_circle: **Title** : Arabic Dates Classification using DL
:red_circle: **Aim** : The objective of this project is to classify images of Arabic Dates into 9 species from the Arabic region.
:red_circle: **Brief Explanation** : The project deals with multi-class classification, classifying images of Arabic Dates into 9 species.

### Screenshots 📷

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Ananya Gupta
- What is your participant role? SSOC

***********************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by SupratikB23**
Closes: #329

### Describe the add-ons or changes you've made 📃
- Performed a PyTorch model training script for Arabic Dates classification (ResNet50, 15 epochs) and saved the model.
- Added Streamlit app for real-time image classification and prediction display.
- Implemented data augmentation & normalization for better generalization.
- Saved trained model to assets/arabic_dates_model.pth.
- Improved model accuracy, dropout, and inference efficiency.

- [X] Bug fix 
- [X] New feature 
- [X] Code style update (formatting, local variables) 
- [X] Documentation update 
- [X] Database update 

### Working Video 📷

https://github.com/user-attachments/assets/0db10129-608f-4619-bf0e-1c2301517876

### CODE REVIEW

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
LGTM!
For GitHub Lint error, It's due to you have not formatted your file in 2-space indentation. Please update this to 2-space.

For any blocker, connect me on discord or ping me here 👍

**Comment by Avdhesh-Varshney** [comment]
@SupratikB23 Could you please gone through the movie recommendations model already integrated in Jarvis?
Because model file cannot be saved in repo this will increase the size of repo.

**Comment by SupratikB23** [comment]
@Avdhesh-Varshney  
I have removed the page configs, created a new Kaggle file for the model creation ( [Kaggle Link](https://www.kaggle.com/code/supratikbhowal/arabic-dates-classification-resnet50) ) and lastly made all the code in 2-indentation format....
When I'm running the specific _**'ArabicDatesClassifierModel.py'**_ file, it's working seamlessly, with Kaggle Notebook fetching and all. But the problem I'm facing is that when I'm running the whole Jarvis program locally, where during authentication it's saying **"400: Missing "authorize_url" value"**. I can't resolve that issue, even though I have followed all the steps like 'secrets.toml' creation, Mock Authentication Creation, etc. Can you please help me with that, else everything's working good

**Comment by SupratikB23** [comment]
Yes done checks with AUTH (video attached), also have made the file to 2-space indentation format.

https://github.com/user-attachments/assets/0a3fe90a-271b-4c13-b257-2da18260ed89

**Review by Avdhesh-Varshney — APPROVED** [review]
Perfect 🎉 
Thanks for your contribution!
